xmppserver/src/main/webapp/session-row.jspf

```diff
@@ -43,10 +43,21 @@
     </td>
     <td style="width: 15%; white-space: nowrap">
         <% if (sess.getSoftwareVersion() != null) {
-            String version = sess.getSoftwareVersion().get("version");
-            if (version != null) { %>
-        <%= StringUtils.escapeHTMLTags(version) %>
-        <% } } %>
+            final String softwareName = sess.getSoftwareVersion().get("name");
+            final String softwareVersion = sess.getSoftwareVersion().get("version");
+
+            String softwareString = "";
+            if(softwareName != null && !softwareName.isBlank()){
+                softwareString += softwareName + " - ";
+            }
+            if(softwareVersion != null && !softwareVersion.isBlank()) {
+                softwareString += softwareVersion;
+            };
```

Comment:
If `name` has a value, but `version` has not, this seems to result in a string that ends with a dash. Maybe do this instead?

```suggestion
            String softwareString = "";
            if(softwareName != null && !softwareName.isBlank()){
                softwareString += softwareName;
            }
            if(softwareVersion != null && !softwareVersion.isBlank()) {
                if (!softwareString.isBlank()) {
                    softwareString += " - ";
                }
                softwareString += softwareVersion;
            };
```

Reply:
Yep, you're absolutely right. I spent so long chasing down an NPE that occurs after the session is established but before the version info is fully available, that I completely missed this.